### PR TITLE
chore: upgrade-18 cherry-picks round 2

### PR DESCRIPTION
**Pull request by mujahidkay**
## Description

Cherry-picks the following commits from master:
 - https://github.com/Agoric/agoric-sdk/pull/10454  (https://github.com/Agoric/agoric-sdk/commit/d16781f0100b831638b3032837da28c23beabb7b)
 - https://github.com/Agoric/agoric-sdk/pull/10455 (https://github.com/Agoric/agoric-sdk/commit/1aec8d05036f6b3c3e3730339d1829da6b4a9051)
 
Also adds a new upgrade name `agoric-upgrade-18-basic-2`. Also, moves provisioning GOV4 from USE to PREPARE.

used `git cherry-pick` for the above commits

 [Draft PR with reapply upgrade on top of rc0](https://github.com/Agoric/agoric-sdk/pull/10484)

### CODE REVIEW

File: golang/cosmos/app/upgrade.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ var upgradeNamesOfThisVersion = []string{
 	"agoric-upgrade-18-devnet",
 	"agoric-upgrade-18-emerynet",
 	"agoric-upgrade-18-basic",
+	"agoric-upgrade-18-basic-2",
```
Comment (mhofman) [inline golang/cosmos/app/upgrade.go:21]: While it doesn't change the behavior, we should have this new name return false in `isPrimaryUpgradeName`

Comment (mhofman) [reply golang/cosmos/app/upgrade.go:21]: I'm also wondering if we should bikeshed the name as it's not really like basic or meant to be applied on its own

**Review by mhofman — COMMENTED** [review]
Is there a draft PR where we can see a reapply of the upgrade on top of rc0 to verify it would fix emerynet?

File: golang/cosmos/app/upgrade.go (hunk 1 above)
Comment (mujahidkay) [reply golang/cosmos/app/upgrade.go:21]: >I'm also wondering if we should bikeshed the name as it's not really like basic or meant to be applied on its own

what do you suggest?

**Comment by mujahidkay** [comment]
>Is there a draft PR where we can see a reapply of the upgrade on top of rc0 to verify it would fix emerynet?

I can work on this today, and then attach it to this PR's description

**Review by Chris-Hibbert — APPROVED** [review]

File: golang/cosmos/app/upgrade.go (hunk 1 above)
Comment (mhofman) [reply golang/cosmos/app/upgrade.go:21]: `agoric-upgrade-18-reapply` ? But honestly we can keep as is too.

**Review by mhofman — APPROVED** [review]
Given the passing tests in #10484, LGTM, but before merging I would like to see the `node ./addGov4` moved to `prepare.sh` to keep verifying in a3p-integration that we have indeed fixed the problem we had in the first place.

**Comment by mujahidkay** [comment]
> Given the passing tests in #10484, LGTM, but before merging I would like to see the `node ./addGov4` moved to `prepare.sh` to keep verifying in a3p-integration that we have indeed fixed the problem we had in the first place.

Good idea.

**Comment by mujahidkay** [comment]
> Given the passing tests in #10484, LGTM, but before merging I would like to see the `node ./addGov4` moved to `prepare.sh` to keep verifying in a3p-integration that we have indeed fixed the problem we had in the first place.

@mhofman Did a run (passing) with this: https://github.com/Agoric/agoric-sdk/actions/runs/11914640186/job/33203061974?pr=10526

**Pull request merged**